### PR TITLE
Python 3: file() --> open() in yaml-highlight example

### PR DESCRIPTION
**Pull request by cclauss**
__file()__ was removed in Python 3 in favor of __open()__ so these changes should be compatible with both Python 2 and Python 3.  Without these changes, this example will not run on Python 3.

### CODE REVIEW

**Comment by cclauss** [comment]
@webknjaz Your review please?

File: examples/yaml-highlight/yaml_hl.py [hunk 1]
```diff
@@ -37,14 +37,14 @@ def __setstate__(self, state):
 class YAMLHighlight:
 
     def __init__(self, options):
-        config = yaml.load(file(options.config, 'rb').read())
+        config = yaml.load(open(options.config, 'rb').read())
```
Comment (webknjaz) [inline examples/yaml-highlight/yaml_hl.py]: Better, but it leaves an open file descriptor hanging... you should either close it or (much better) use a context manager:
```suggestion
        with open(options.config, 'rb').read() as yaml_file:
            config = yaml.load(yaml_file.read())
```

File: examples/yaml-highlight/yaml_hl.py [hunk 2]
```diff
@@ -37,14 +37,14 @@ def __setstate__(self, state):
...
         self.style = config[options.style]
         if options.input:
-            self.input = file(options.input, 'rb')
+            self.input = open(options.input, 'rb')
```
Comment (webknjaz) [inline examples/yaml-highlight/yaml_hl.py]: Is this one and the next one closed somewhere?

Comment (webknjaz) [reply examples/yaml-highlight/yaml_hl.py]: (at least in `__del__` or so)

Comment (webknjaz) [reply examples/yaml-highlight/yaml_hl.py]: Of course, it's much better and safer to turn this example into a context manager handling this properly on exit...

Comment (webknjaz) [reply examples/yaml-highlight/yaml_hl.py]: Or accept an open file-like object as an option and handle open/close via CM on that level.

**Comment by cclauss** [comment]
Bump on this one.

File: examples/yaml-highlight/yaml_hl.py (hunk 2 above)
Comment (cclauss) [reply examples/yaml-highlight/yaml_hl.py]: Does this change resolve it?

**Comment by cclauss** [comment]
@hugovk Your review please

**Comment by perlpunk** [comment]
I'm getting the following error with python 3.6.4:
```
% PYTHONPATH=$PWD/lib3 python3 examples/yaml-highlight/yaml_hl.py <.appveyor.yml
Traceback (most recent call last):
  File "examples/yaml-highlight/yaml_hl.py", line 126, in <module>
    hl.highlight()
  File "examples/yaml-highlight/yaml_hl.py", line 66, in highlight
    if input.startswith(codecs.BOM_UTF16_LE):
TypeError: startswith first arg must be str or a tuple of str, not bytes
```

**Comment by perlpunk** [comment]
Thanks! I merged it to release/5.3 although there seem to be more changes necessary like I mentioned above.

**Comment by perlpunk** [comment]
I'm also getting a warning in python2:
```
Exception AttributeError: "'NoneType' object has no attribute 'stdin'" in <bound method YAMLHighlight.__del__ of <__main__.YAMLHighlight instance at 0x7fb9b81a47a0>> ignored 
```
I think I should revert

File: examples/yaml-highlight/yaml_hl.py [hunk 3]
```diff
@@ -37,17 +43,24 @@ def __setstate__(self, state):
 class YAMLHighlight:
 
     def __init__(self, options):
-        config = yaml.load(file(options.config, 'rb').read())
+        with open(options.config, 'rb') as yaml_file:
+            config = yaml.load(yaml_file.read())
```
Comment (cclauss) [inline examples/yaml-highlight/yaml_hl.py]: __yaml__ —> __yaml_file__ on this line.

**Comment by perlpunk** [comment]
Reverted

**Pull request merged**